angr/analyses/cfg/cfg_fast.py

```diff
@@ -3803,7 +3811,9 @@ def _generate_cfgnode(self, cfg_job, current_function_addr):
                 # is there a section?
                 has_executable_section = self._object_has_executable_sections(obj)
                 section = obj.find_section_containing(addr)
-                if has_executable_section and section is None:
+                # is there a segment?
+                segment = obj.find_segment_containing(addr)
```

Comment:
If there is already a section, do we need to run `find_segment_containing(addr)` any more? I feel this line should be
```
if section is None:
    segment = obj.find_segment_containing(addr)
```
Sorry for being picky, but this function is pretty hot during CFG recovery, and we want to do as little redundant work as possible.

Reply:
Sure, I understand that. As I mentioned at the beginning, it's just a dirty patch and I'm very willing to improve it :)

And yes, I agree with you. Similarly, do you think we need to add `self._object_has_executable_segments` and check for `has_executable_segment` before `obj.find_segment_containing(addr)` ?